sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/util/PassThroughWindowFn.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.dataflow.sdk.util;
+
+import com.google.cloud.dataflow.sdk.Pipeline;
+import com.google.cloud.dataflow.sdk.coders.Coder;
+import com.google.cloud.dataflow.sdk.transforms.windowing.BoundedWindow;
+import com.google.cloud.dataflow.sdk.transforms.windowing.NonMergingWindowFn;
+import com.google.cloud.dataflow.sdk.transforms.windowing.WindowFn;
+
+import org.joda.time.Instant;
+
+import java.util.Collection;
+
+/**
+ * A {@link WindowFn} that leaves all associations between elements and windows unchanged.
+ *
+ * <p>In order to implement all the abstract methods of {@link WindowFn}, this requires the
```

Comment:
This is not a delegate WindowFn, especially now that the priorWindowFn has been removed. It cannot be, as to be applied via `Window.into(WindowFn)` it must be able to provide a meaningful result for assignWindows, and is used to apply a GroupByKey where the input could have a windowing strategy that is non idempotent (or is in InvalidWindows, etc)

we must be able to apply this WindowFn, which means it must have an `assignWindows` that functions.